pkg/cmd/pulumi/state.go

```diff
@@ -177,7 +177,7 @@ func totalStateEdit(ctx context.Context, s backend.Stack, showPrompt bool, opts
 	}
 
 	// If the stack is already broken, don't bother verifying the integrity here.
-	if !stackIsAlreadyHosed {
+	if !stackIsAlreadyHosed && !backend.DisableIntegrityChecking {
```

Comment:
I wonder if this should be here though.  Since it is a bug if we fail to verify the integrity here, should `DisableIntegrityChecking` work around that?  I kinda think this is the right thing to do still, but it's the one place where I'm a little bit unsure.

Reply:
Yeh I mean _really_ VerifyIntegrity should _never_ fail, but it does and DisableIntegrityChecking is how you work around that. So if we didn't check it here you'd get a panic after the state edit and wouldn't be able to save it even if you really wanted to.

Reply:
Makes sense!